File: 221228-dojo/java/app/src/main/java/Solution.java
class Solution {
    public int[] solution(int n, int m) {
        int smallNumber = n;

        if (smallNumber > m) {
            smallNumber = m;
            m = n;
        }

        int gcd = getGcd(smallNumber, m);

        int lcm = getLcm(gcd, smallNumber, m);

        return new int[]{gcd, lcm};
    }

    public int getGcd(int smallNumber, int m) {
        for (int i = smallNumber; i > 0; i -= 1) {
            if (smallNumber % i == 0 && m % i == 0) {
                return i;
            }
        }
        return 1;
    }

    public int getLcm(int gcd, int smallNumber, int m) {
        if (m % smallNumber == 0) {
            return m;
        }

        int accumulator = 1;

        while (true) {
            if (gcd > 1 && smallNumber % gcd == 0 && m % gcd == 0) {
                accumulator *= gcd;
                smallNumber /= gcd;
                m /= gcd;
                continue;
            }

            return accumulator * smallNumber * m;
        }
    }
}

File: 221228-dojo/javascript/solution.js
export function getGcd(small, large, divisor) {
  if (large % divisor === 0 && small % divisor === 0) {
    return divisor;
  }

  return getGcd(small, large, divisor - 1);
}

export function getLcm(small, large, gcd) {
  return (small * large) / gcd;
}

export default function solution(n, m) {
  const small = Math.min(n, m);
  const large = Math.max(n, m);
  const divisor = small;

  const gcd = getGcd(small, large, divisor);
  const lcm = getLcm(small, large, gcd);

  return [gcd, lcm];
}
